onnxruntime/core/optimizer/pre_shape_node_elimination.cc

```diff
@@ -0,0 +1,75 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+#include "core/common/logging/logging.h"
+#include "core/optimizer/rewrite_rule.h"
+#include "core/optimizer/pre_shape_node_elimination.h"
+#include "core/optimizer/utils.h"
+#include "core/graph/graph.h"
+#include "core/graph/graph_utils.h"
+
+namespace onnxruntime {
+
+Status PreShapeNodeElimination::Apply(Graph& graph, Node& node, RewriteRuleEffect& rule_effect, const logging::Logger&) const {
+  const Node* p_input_node = graph_utils::GetInputNode(node, 0);
```

Comment:
we don't need get the p_input_node, this is only to get the Cast's node's input_arg, why not just use node.MutableInputDefs[0]